src/sql/base/browser/ui/panel/panel.ts

```diff
@@ -111,9 +112,10 @@ export class TabbedPanel extends Disposable {
 		return this._tabMap.has(tab.identifier);
 	}
 
-	public pushTab(tab: IPanelTab, index?: number): PanelTabIdentifier {
+	public pushTab(tab: IPanelTab, index?: number, containsWebView?: boolean): PanelTabIdentifier {
```

Comment:
should this be something more generically named, like "destroy on removal" or something.

Reply:
sure, this parameter will change in the next iteration anyways, since it really needs to be a tristate setting, `default`, `rebuild`, or `retainWhenHidden`.  Do you have have a suggestion of how best to capture that?  Though as mentioned retain will come separately.

Reply:
what would retainwhenhidden do? I feel like most of this logic should be baked into the tabs themselves rather than into the panel. might make more sense to have a "onremoval" or "ondestroy" optional function on the tabs themselves which would handle performing this logic.

Reply:
@anthonydresser let me rename to destroy and push this in.  Then we can iterate on that design in the next PR which will add the retainWhenHidden support.  This unblocks a couple people waiting for this.